tools/report-converter/codechecker_report_converter/analyzers/analyzer_result.py

```diff
@@ -121,12 +121,19 @@ def _post_process_result(self, reports: List[Report]):
         By default it will add report hashes and metada information.
         """
         for report in reports:
-            self._add_report_hash(report)
             self._add_metadata(report)
+            self._add_report_hash(report)
 
     def _add_report_hash(self, report: Report):
         """ Generate report hash for the given plist data. """
-        report.report_hash = get_report_hash(report, HashType.CONTEXT_FREE)
+        hash_type = HashType.CONTEXT_FREE
+
+        # Due to backward compatibility, the CodeChecker analyzer
+        # uses hash type PATH_SENSITIVE for ClangTidy by default.
+        if report.analyzer_name == "clang-tidy":
```

Comment:
This should rather be implemented in an override method in `tools/report-converter/codechecker_report_converter/analyzers/clang_tidy/analyzer_result.py`.

Also, don't forget to change `clang_tidy_yaml`.